### PR TITLE
Ingredients for layer unit tests

### PR DESCRIPTION
**Pull request by timmoon10**
This PR makes progress toward a simple framework for layer unit testing (#624):
- Implemented a "value initializer", a weights initializer that populates entries based on a user-provided list of values.
- Implemented a "weights layer" that outputs its weights tensor.
- Implemented an L2 norm squared layer on CPU and GPU.
- Removed the "L2 loss" layer, which had a somewhat misleading name (I blame TensorFlow) and was implemented as an activation layer. I've gone ahead and made changes to the affected models (mostly CycleGAN models).
- Added a unit test for MNIST softmax classification. The prototext had already been written, but it wasn't registered with Bamboo.
- Added a unit test for the L2 norm squared layer. This is a template for future layer unit tests.
- The LBANN executables now exit with a failure status if an exception is caught. PyTest was previously reporting false successes.

The unit tests pass when I run manually on Pascal and Catalyst, but I'd like to see how they perform under Bamboo.

### CODE REVIEW

**Comment by timmoon10** [comment]
Note that a combination of the weights layer, value initializer, and synthetic data reader may be helpful for creating a deterministic, self-contained smoke test (#53). The limitation is that each mini-batch sample will be identical.
